File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character,Integer> mapS = new HashMap<>(),mapT = new HashMap<>();
        int n = s.length();
        int m = t.length();
        if(n != m) return false;
        for(int i=0;i<n;i++){
            mapS.put(s.charAt(i),mapS.getOrDefault(s.charAt(i),0)+1);
            mapT.put(t.charAt(i),mapT.getOrDefault(t.charAt(i),0)+1);
        }
        if(mapS.equals(mapT)) return true;
        return false;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> mapS,mapT;
        int n = s.size();
        int m = t.size();
        if(n != m) return false;
        for(int i=0;i<n;i++){
            mapS[s[i]]++;
            mapT[t[i]]++;
        }
        if(mapS == mapT) return true;
        return false;

    }
};